api/python/provisioner/commands/configure_setup.py

```diff
@@ -90,199 +98,166 @@ def __attrs_post_init__(self):
-        logger.debug(f"params data {content}")
-
-        input_type = None
-        pillar_type = None
-        node_list = []
-        count = int(number_of_nodes)
-
-        for section in content:
-            input_type = section
-            pillar_type = section
+        logger.info("Updating salt data..")
+        content = {section: dict(config.items(section))
+                   for section in config.sections()}  # noqa: E501
+
+        logger.debug(
+            f"Config data read from provided file: {content}"
+        )
+
+        # Config sections segregated default-wise, node-wise.
+        parsed_nodes = validate._parse_nodes(content)
+
+        pillar_map = {f'{NODE_DEFAULT}': f'{CLUSTER}',
+                      f'{NODE}': f'{CLUSTER}',
+                      f'{CLUSTER}': f'{CLUSTER}',
+                      f'{STORAGE}': f'{STORAGE}',
+                      f'{STORAGE_DEFAULT}': f'{STORAGE}'}
+
+        for section in parsed_nodes['section_list']:
             if 'srvnode' in section:
-                input_type = 'node'
-                pillar_type = f'cluster/{section}'
-                count = count - 1
-                node_list.append(f"\"{section}\"")
+                input_type = (f'{NODE_DEFAULT}' if 'default' in section
+                              else f'{NODE}')
+            elif 'storage' in section:
+                input_type = (f'{STORAGE_DEFAULT}' if 'default' in section
+                              else f'{STORAGE}')
+
+            elif 'cluster' in section:
+                input_type = f'{CLUSTER}'
+
+            logger.debug(
+               f"Validating Params in section: {section}"
+            )
 
             self._validate_params(input_type, content[section])
             self._parse_input(content[section])
 
-            for pillar_key in content[section]:
-                key = f'{pillar_type}/{self._parse_pillar_key(pillar_key)}'
-                run_subprocess_cmd([
-                       "provisioner", "pillar_set",
-                       key, f"{content[section][pillar_key]}"])
+            pillar_map_type = pillar_map[input_type]
 
-        if content.get('cluster', None):
-            if content.get('cluster').get('cluster_ip', None):
-                run_subprocess_cmd([
-                       "provisioner", "pillar_set",
-                       "s3clients/ip",
-                       f"{content.get('cluster').get('cluster_ip')}"])
+            # Contents of 'default' sections will be applied to each node
+            if input_type == f'{NODE_DEFAULT}':
+                for each_node in parsed_nodes['server_list']:
+                    pillar_type = f'{pillar_map_type}/{each_node}'
+                    self._update_pillar_data(content, section, pillar_type)
 
-        if count > 0:
-            raise ValueError(f"Node information for {count} node missing")
+            elif input_type == f'{STORAGE_DEFAULT}':
```

Comment:
The same concern as for node: update with defaults should go first before any specific set of values. Please ensure that it is always true.

Reply:
Hope this block ensure that. Once the setup is up and running, will test on multi-node setups and update test scenarios accordingly.